### PR TITLE
Adding service monitor

### PR DESCRIPTION
**Pull request by ashwinrajadurai3**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | nil
| License         | Apache 2.0

### What's in this PR?
This PR adds service monitor object for spot-termination-exporter helm-chart

### Why?

To monitor this service from prometheus operator we need this added in template

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

**Comment by ashwinrajadurai3** [comment]
Can we review and merge this PR? @tarokkk @pregnor @mpritter76 @swirsky

**Review by pregnor — COMMENTED** [review]
I have reviewed the changes from a general perspective, because I was explicitly requested for some reason unknown to me.

However I have no contextual knowledge about the `spot-termination-exporter` chart or the component it packs and thus of the changes which also means I'm not sure whether this feature is desired from the higher level chart development perspective and if so in what manner.

Therefore I'm passing the chance of review onto the experts of the chart, whoever them may be.

File: spot-termination-exporter/values.yaml
Comment (ashwinrajadurai3) [reply spot-termination-exporter/values.yaml]: We use additionalLabels & jobLabels hence removed rest as suggested, Thanks

File: spot-termination-exporter/values.yaml
Comment (ashwinrajadurai3) [reply spot-termination-exporter/values.yaml:38]: Thanks for your review, Appreciate it. Yeah i agree on that but this particular exporter needs prometheus deployment feature enabled as a default policy for it's functionality. 
Can we have this particular case as an exception? If you are not convinced i will look on an other way around, please let me know

Comment (pregnor) [reply spot-termination-exporter/values.yaml:38]: Sure, if that's the case it is fine the way it is (as an exception with the default being enabled).

File: spot-termination-exporter/values.yaml
Comment (pregnor) [reply spot-termination-exporter/values.yaml]: Note: we don't use the `jobLabel` though, the service monitor's job label is dynamically set to the release name [here](https://github.com/banzaicloud/banzai-charts/pull/1271/files#diff-83c04ea7e8d711d785a57269414df521b5f63570bdf865eefcc6cdec7343a3fbR17).

Comment (ashwinrajadurai3) [reply spot-termination-exporter/values.yaml]: Yeah removed it, can you please approve now?

Comment (pregnor) [reply spot-termination-exporter/values.yaml]: As I said, LGTM, but I'm not the component/chart maintainer thus I think my approval would convey a false message, so I would rather let the approval of the PR to the real maintainers.

**Review by pregnor — COMMENTED** [review]
-

**Comment by pregnor** [comment]
@ahma could you take a look at this when you have the time?

**Review by pregnor — APPROVED** [review]

**Comment by ashwinrajadurai3** [comment]
> I'm going to approve the changes as they seem all right to me, but I'm also going to make sure to have it reviewed by a maintainer of the chart as well, because I'm not a maintainer of the chart.

Thanks for that @pregnor

**Comment by ashwinrajadurai3** [comment]
> @ahma could you take a look at this when you have the time?

@ahma Let me know if this looks good to you, thanks!

**Comment by ashwinrajadurai3** [comment]
@tarokkk @ahma @mpritter76 @swirsky
May i please request one of you to take a look at this PR.

**Comment by ashwinrajadurai3** [comment]
@pepov @sancyx  @khernyo @12ushan @asdwsda
May i please request one of you to take a look at this PR or let me know who can do the needful

**Comment by pepov** [comment]
I would also say that the servicemonitor should be disabled by default.

**Comment by ashwinrajadurai3** [comment]
> I would also say that the servicemonitor should be disabled by default.

I concur that @pepov I've modified the changes accordingly. Can i please get an approval and merge now, Thanks!

**Review by pepov — APPROVED** [review]

**Pull request merged**
